product-scenarios/scenarios-commons/src/main/java/org/wso2/carbon/esb/scenario/test/common/testng/listeners/TestExecutionListener.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.scenario.test.common.testng.listeners;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.awaitility.Awaitility;
+import org.testng.IExecutionListener;
+import org.wso2.carbon.application.mgt.stub.ApplicationAdminExceptionException;
+import org.wso2.carbon.esb.scenario.test.common.AuthenticatorClient;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioConstants;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.wso2.carbon.integration.common.admin.client.ApplicationAdminClient;
+import org.wso2.carbon.integration.common.admin.client.CarbonAppUploaderClient;
+
+import java.io.File;
+import java.rmi.RemoteException;
+import java.util.Properties;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import javax.activation.DataHandler;
+import javax.activation.FileDataSource;
+
+/**
+ * TestNG execution listener for perform test execution preparation
+ * - deploy carbon applications if not already deployed
+ */
+public class TestExecutionListener implements IExecutionListener {
+
+    private static final Log log = LogFactory.getLog(TestExecutionListener.class);
+    private boolean standaloneMode = false;
+    private String backendURL;
+    private String mgtConsoleUrl;
+    private String sessionCookie;
+
+    private CarbonAppUploaderClient carbonAppUploaderClient = null;
+    private ApplicationAdminClient applicationAdminClient = null;
+
+    private static final String resourceLocation = System.getProperty(ScenarioConstants.TEST_RESOURCES_DIR);
+    private static final String carbonApplicationsDir =
+            System.getProperty(ScenarioConstants.TEST_RESOURCES_CARBON_APPLICATIONS_DIR);
+
+    private Properties infraProperties;
+
+    @Override
+    public void onExecutionStart() {
+
+        infraProperties = ScenarioTestBase.getDeploymentProperties();
+        // TODO : uncomment this once test environment is stable and remove hardcoded value
+        //standaloneMode = Boolean.valueOf(infraProperties.getProperty(ScenarioConstants.STANDALONE_DEPLOYMENT));
+        standaloneMode = true;
+
+        if (standaloneMode) {
+            log.info("Test execution standalone mode : " + standaloneMode);
```

Comment:
the log is printed inside if (standaloneMode) hence, it will always be true

Reply:
it is temporary till test grid capp deployment finalized. Already added TODO comment few lines above